### ISSUE TITLE
PDF: "goto page" vs. reported page number discrepancy (i.e. page number indicated at bottom of A4 page vs. number of A4 individual pages)

### ISSUE THREAD

**Issue by danielweck** [edrlab/thorium-reader#1340]
To reproduce:
1) Open "Anatomy Of An Election" (our usual test PDF)
2) Goto page "3"
3) scroll down to see that the page number is indeed "3" (visually displayed at the bottom of the PDF page)
4) notice that page "32" is in fact reported in the console, and near the "goto page" input control, and there are indeed 30+ pages "above" the current page in the scroll view

I am implementing got begin/end of publication (quick shortcuts), and I noticed this bug in the PDF notion of "pages" (I need to be able to invoke goto(1) / first page of the publication)

**Comment by danielweck** [comment]
This works fine with PDF.js sample "Trace-based Just-in-Time Type Specialization for DynamicLanguages" ("compressed.tracemonkey....pdf"), because the logical page numbers match the physical ones.

**Comment by danielweck** [comment]
@panaC 
> the first page isn't always "1" ans could be any string.
seems to me

Sure, but the page number event reported in the console and displayed in the "goto page" UI indicates that there is a [1-n] integer sequence available from PDF.js, regardless of textual / symbolic "page markers". See my description above (the discrepancy is obvious, and easy to reproduce).

**Comment by panaC** [comment]
I don't know how to resolve the bug , the same bug exists in the footer reading bar

**Comment by danielweck** [comment]
It seems the PDF "engine" conflates physical and logical page numbers. There seems to be a discrepancy between the "go to" API, and the "where am I" API.

**Comment by panaC** [comment]
I'm closing this issue. feel free to reopen it

**Comment by danielweck** [comment]
I agree about closing this issue as "cannot fix" because the `pageNumber` and `pageChange` events from PDF.js report different numbers than the "goto page" action. To summarise (I have just tested this again with the PDF mentioned in my original comment): from the user perspective, "goto page" works fine as the page indicator displayed at bottom of the PDF page matches that entered by the user. However the page number reported by the locator events is incorrect (displayed in the bottom progression bar, as well as immediately next to the "goto page" user interface)

**Comment by panaC** [comment]
I'm reopening the issue, the issue always remain broken in 2024 and thorium 3.1.

**Comment by danielweck** [comment]
We need to implement a width threshold (minimum value in CSS pixels) for the "spans" in the bottom navigation bar, and skip rendering HTML elements that are too small visually / for hit testing with the mouse cursor. This will introduce gaps in the sequence of pages (or spine items, same problem with EPUBs etc.), but this is an acceptable tradeoff. In fact, the current rendering model doesn't allow pointing at individual segments anyway, so this will be a net benefit in terms of performance, no usability loss.

**Comment by danielweck** [comment]
I am posting a separate issue for the bottom bar overload.
